includes/admin/importers/class-wc-product-csv-importer-controller.php

```diff
@@ -395,44 +395,51 @@ protected function get_default_fields() {
 	protected function auto_map_columns( $raw_headers, $num_indexes = true ) {
 		$weight_unit     = get_option( 'woocommerce_weight_unit' );
 		$dimension_unit  = get_option( 'woocommerce_dimension_unit' );
-		$default_columns = array_flip( apply_filters( 'woocommerce_csv_product_import_mapping_default_columns', array(
```

Comment:
I had to ditch the `array_flip` thing because it prevented having e.g. two possible things mapping to 'id'. 

Having all of the automatic mappings in one big array is going to give us the best performance since you can just check if the index exists for each column like we do on line 464.